.github/workflows/e2e-ci.yml

```diff
@@ -33,7 +32,8 @@ jobs:
       -
         uses: actions/setup-go@v4
         with:
-          go-version: ${{ env.SETUP_GO_VERSION }}
+          go-version-file: 'go.mod'
```

Comment:
For information, @thardeck reverted a similar change I did on the rancher repository 
https://github.com/rancher/fleet/pull/1761#discussion_r1318226234

Reply:
Thanks for the comment! I believe that is actually a corner case, since it interacts with the rancher repository, which is still using an older version in its `go.mod` file. Please note that there are other workflows that use the `go-setup` action but only `release-against-rancher.yml` had to be reverted.

Since this repository does not suffer from that restriction, I think it's okay to extract the Go version from `go.mod`.